test/Registry.ts

```diff
@@ -140,14 +140,12 @@ describe(REGISTRY_CONTRACT_NAME, function () {
 
       // Non-owner (addr1) should not be able to register the referrer
       await expect(
-        registry
-          .connect(addr1)
-          .registerReferrer(
-            mockReferrerIdHex,
-            protocolIds,
-            mockRewardRates,
-            mockRewardAddress,
-          ),
+        (registry.connect(addr1) as typeof registry).registerReferrer(
```

Comment:
Ideally we wouldn't need these casts.

However this is because we use `ethers` in those tests, because of `hardhat-upgrades` which uses it (see https://github.com/valora-inc/valora-points-contracts/pull/5#discussion_r1559275283, and https://github.com/OpenZeppelin/openzeppelin-upgrades/issues/1071).

And it requires `typechain` to get proper types, which we could definitely do, but maybe there's a way to wrap with viem?